#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_hurd_detection.dpatch by Cyril Brulebois <cyril.brulebois@enst-bretagne.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Enable GNU/Hurd detection

@DPATCH@
diff -Nur qt4-x11-4.1.3~/src/corelib/global/qglobal.h qt4-x11-4.1.3~/src/corelib/global/qglobal.h
--- qt4-x11-4.1.3~/src/corelib/global/qglobal.h	2006-07-21 11:51:19.000000000 +0000
+++ qt4-x11-4.1.3/src/corelib/global/qglobal.h	2006-07-21 11:58:25.000000000 +0000
@@ -69,6 +69,7 @@
      LYNX     - LynxOS
      BSD4     - Any BSD 4.4 system
      UNIX     - Any UNIX BSD/SYSV system
+     HURD     - Hurd
 */
 
 #if defined(__APPLE__) && (defined(__GNUC__) || defined(__xlC__) || defined(__xlc__))
@@ -136,6 +137,8 @@
 #  define Q_OS_UNIXWARE
 #elif defined(__svr4__) && defined(i386) /* Open UNIX 8 + GCC */
 #  define Q_OS_UNIXWARE
+#elif defined(__GNU__) /* To avoid confusion, like for mkspecs files */
+#  define Q_OS_HURD
 #elif defined(__MAKEDEPEND__)
 #else
 #  error "Qt has not been ported to this OS - talk to qt-bugs@trolltech.com"
